### PR TITLE
Add InitiateMaintenance New Command

### PR DESCRIPTION
**Pull request by vanbasten2323**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj [hunk 1]
```diff
@@ -36,6 +36,7 @@
     <None Include="OneWebOneWorker.cspkg">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="packages.config" />
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj]: I don't think you need these changes. Probably generated by VS or the build process. Please try to revert them.

File: src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj [hunk 2]
```diff
@@ -172,11 +173,38 @@
       <Name>ComputeManagement</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.Threading.Tasks, Version=1.0.12.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj]: same as above

File: src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj [hunk 3]
```diff
@@ -172,11 +173,38 @@
...
+    </Reference>
+  </ItemGroup>
   <Target Name="AfterBuild">
     <PropertyGroup>
       <PowerShellExe Condition=" '$(PowerShellExe)'=='' ">%WINDIR%\System32\WindowsPowerShell\v1.0\powershell.exe</PowerShellExe>
       <MockSessionRecordScriptPath>"$(OutputPath)\MockSessionRecords.ps1"</MockSessionRecordScriptPath>
     </PropertyGroup>
     <Exec Command="$(PowerShellExe) $(MockSessionRecordScriptPath)" />
   </Target>
+  <Import Project="..\..\..\..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets" Condition="Exists('..\..\..\..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets')" />
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj]: same as above

File: src/TestDependencies/packages.config [hunk 4]
```diff
@@ -1,30 +1,28 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="FakeItEasy" version="1.22.0" targetFramework="net451" />
   <package id="Hyak.Common" version="1.0.2" targetFramework="net451" />
   <package id="Microsoft.AspNet.WebApi.Client" version="5.2.2" targetFramework="net451" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net451" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net451" />
+  <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net451" />
```
Comment (huangpf) [inline src/TestDependencies/packages.config]: Why do you need these changes?

File: src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj (hunk 2 above)
Comment (markcowl) [inline src/ServiceManagement/Compute/Compute.Tests/Compute.Tests.csproj]: This dependency should already include all the files you need: https://github.com/Azure/azure-sdk-for-net/pull/2957/files#diff-27bb0edb25bb7f42bd18d1985d5ff50bR167 .  This may be the cause of build failures.

File: src/ServiceManagement/Compute/ComputeManagement/ComputeManagement.csproj [hunk 5]
```diff
@@ -32,9 +33,18 @@
     <Reference Include="Microsoft.Azure.Common">
       <HintPath>$(LibraryNugetPackageFolder)\Microsoft.Azure.Common.2.0.4\lib\portable-net45+wp8+wpa81+win\Microsoft.Azure.Common.dll</HintPath>
     </Reference>
-    <Reference Include="System.Threading.Tasks" />
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/ServiceManagement/Compute/ComputeManagement/ComputeManagement.csproj]: You should not need to make changes to this csproj, other than including any new generated files

File: src/TestDependencies/packages.config [hunk 6]
```diff
@@ -1,30 +1,28 @@
...
+  <package id="Microsoft.Azure.Management.Resources" version="2.18.0-preview" targetFramework="net45" />
```
Comment (markcowl) [inline src/TestDependencies/packages.config]: Why is this needed?

**Review by markcowl — CHANGES_REQUESTED** [review]
Please read and fill out the checklist.  This review just encompasses the two files most likely to be causing your build problems.

You can likely reproduce build problems locally by cleaning your local clone of the repo (git clean -xdf, after comm itting any changes), and then building using msbuild build.proj from the repo root.

**Comment by hyonholee** [comment]
LGTM.  Thanks.

**Comment by vanbasten2323** [comment]
@markcowl , @amarzavery, @balajikris , would you mind helping me review this PR. Our code freeze date is March 29. Only after this PR is merged can I start developing the powershell project. Thank you!

File: src/TestDependencies/packages.config [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (markcowl) [inline src/TestDependencies/packages.config]: Please revert changes to this file.

File: src/ServiceManagement/Compute/ComputeManagement/packages.config [hunk 8]
```diff
@@ -8,4 +8,4 @@
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="portable-net45+sl50+wp80+win" />
   <package id="Microsoft.Net.Http" version="2.2.22" targetFramework="portable-net45+sl50+wp80+win" />
   <package id="Newtonsoft.Json" version="6.0.4" targetFramework="portable-net45+sl50+win+wp80" />
-</packages>
```
Comment (markcowl) [inline src/ServiceManagement/Compute/ComputeManagement/packages.config]: Please revert changes to this file

File: src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -19,8 +19,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Compute Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Virtual Machines and Hosted Services.")]
 
-[assembly: AssemblyVersion("13.0.0.0")]
-[assembly: AssemblyFileVersion("13.0.0.0")]
+[assembly: AssemblyVersion("14.0.0.0")]
```
Comment (markcowl) [inline src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs]: Are there breakign changes in this update?  If the change is only new API surface, then the new AssemblyFileVersion should be 13.1.0.0 and the AssemblyVersion should be unchanged.

File: src/ServiceManagement/Compute/ComputeManagement/Microsoft.WindowsAzure.Management.Compute.nuget.proj [hunk 10]
```diff
@@ -5,7 +5,7 @@
     Microsoft.WindowsAzure.Management.Compute
     -->
     <SdkNuGetPackage Include="Microsoft.WindowsAzure.Management.Compute">
-      <PackageVersion>13.0.0</PackageVersion>
+      <PackageVersion>14.0.0</PackageVersion>
```
Comment (markcowl) [inline src/ServiceManagement/Compute/ComputeManagement/Microsoft.WindowsAzure.Management.Compute.nuget.proj]: If there were only additions to the API, this should be 13.1.0

**Review by markcowl — CHANGES_REQUESTED** [review]
A couple of minor changes, otherwise looks good.

File: src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (vanbasten2323) [inline src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs]: I am not sure about what you meant by "API surface". I added a new command "initiateMaintenance" and I also added a new field into the response of GetDeployment. Does that count as "new API surface"?

Comment (huangpf) [inline src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs]: API version header is changed, so it's a significant change... @markcowl

File: src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json [hunk 11]
```diff
@@ -13,7 +13,7 @@
           "214"
         ],
         "x-ms-version": [
-          "2016-06-01"
+          "2017-01-01"
```
Comment (huangpf) [inline src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json]: API version change is significant, so the major version shall be upgrade.

File: src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (markcowl) [inline src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs]: significant change doesn't matter, it is not a breaking change, so the version update should be a minor version

File: src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json (hunk 11 above)
Comment (markcowl) [inline src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json]: significant does not matter, it is not a breaking change in the api

File: src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (huangpf) [inline src/ServiceManagement/Compute/ComputeManagement/Properties/AssemblyInfo.cs]: But the server side behavior may be different in terms of exception messages, throttle policy, etc. It's hard to conclude that it's all the same as the previous version.

**Comment by shahabhijeet** [comment]
@vanbasten2323 if this PR also has Powershell changes also, code freeze is 3/29/2017.
Our SLA for PR review is 2 days. If this PR is not resolved and ready for merge by eod Monday (3/27/2017), it's likely your PowerShell PR will not make it to the release.
If you do not have any PS changes, ignore this comment.

Our guidance for major version bump is very clear, breaking changes are prime candidate of major version bump. Do you think you want your SDK users to do major upgrade because service API version changed?
All our users across RP have a notion that major version upgrade are indication of breaking changes.

File: src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json (hunk 11 above)
Comment (markcowl) [inline src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json]: talked to pengfei offline, since there may be behavioral changes with the new rdfe api version we will go with the major version update

Comment (vanbasten2323) [inline src/ServiceManagement/Compute/Compute.Tests/SessionRecords/Microsoft.WindowsAzure.Management.Compute.Testing.ComputeOperationsTests/CanAbortVirtualMachineDeployment.json]: Many thanks to @markcowl  and @huangpf for settling this version issue.

**Comment by vanbasten2323** [comment]
I resolved all the comments. @markcowl  Thank you.

**Review by markcowl — APPROVED** [review]
